File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/util/logger/ILogWriter.java
package com.dkellycollins.triggerfinger.util.logger;

public interface ILogWriter {

    void logDebug(String message);

    void logDebug(String message, Throwable exception);

    void logInfo(String message);

    void logWarn(String message);

    void logWarn(String message, Throwable exception);

    void logError(String message, Throwable exception);

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/util/logger/ILogWriter.kt
package com.dkellycollins.triggerfinger.util.logger

interface ILogWriter {

    fun logDebug(message: String)

    fun logDebug(message: String, exception: Throwable)

    fun logInfo(message: String)

    fun logWarn(message: String)

    fun logWarn(message: String, exception: Throwable)

    fun logError(message: String, exception: Throwable)

}
